### ISSUE TITLE
Locale LC_NUMERIC set decimal point to comma corrupts compilation of assets

### ISSUE THREAD

**Issue by wojtek-robotecai** [o3de/o3de#14836]
**Describe the bug**
Setting LC_NUMERIC for en_DK.UTF-8 (or other which sets decimal point to comma ",", not dot ".")  will affect compilation of assets
and value "0.7" is changed to "0,0" which is treat as "0" .
It affects DynamicFriction , so that wheels are very slippery.

**Assets required**
[o3de/o3de-extras] (https://github.com/o3de/o3de-extras)
[o3de /ROSConDemo] (https://github.com/o3de/ROSConDemo)
may use [dockerfile] (https://github.com/o3de/ROSConDemo/tree/development/docker)
and build `docker build -t roscon_demo -f Dockerfile . `

**Steps to reproduce**
1. Set locale to some that have `locale -k decimal_point` equal to ","
```
locale-gen en_DK.UTF-8
export LC_ALL=en_DK.UTF-8
locale -k decimal_point  #print what is decimal point
```

2. Remove compiled assets:
```
rm -r ROSConDemo/Project/Cache
rm -r ROSConDemo/Project/user
```

3. Compile assets
```
cmake --build ROSConDemo/Project/build/linux --config profile --target ROSConDemo.Assets
```

**Expected behavior**
in `ROSConDemo/Project/Cache/linux/materials/wheel_material.physicsmaterial`
DynamicFriction should be value 0.700:
```
...
<Class name="AZStd::pair&lt;AZStd::basic_string&lt;char, AZStd::char_traits&lt;char&gt;, allocator&gt;, Physics::MaterialPropertyValue&gt;" field="element" type="{04044E5E-35EB-54EF-A4E0-C8785C9233D9}">
  <Class name="AZStd::string" field="value1" value="DynamicFriction" type="{03AAAB3F-5C47-5A66-9EBC-D5FA4DB353C9}"/>
  <Class name="Physics::MaterialPropertyValue" field="value2" version="1" type="{2D9822B6-B4AD-4635-87FD-08F857EEB152}">
    <Class name="float" field="Value" value="0.7000000" type="{EA2C3E90-AFBE-44D4-A90D-FAAF79BAF93D}"/>
  </Class>
</Class>
...
```
which is product of processing:
`o3de-extras/Gems/ROS2/Assets/Materials/wheel_material.physxmaterial`

**Actual behavior**
in `ROSConDemo/Project/Cache/linux/materials/wheel_material.physicsmaterial`
DynamicFriction is value 0,000:
```
...
<Class name="AZStd::pair&lt;AZStd::basic_string&lt;char, AZStd::char_traits&lt;char&gt;, allocator&gt;, Physics::MaterialPropertyValue&gt;" field="element" type="{04044E5E-35EB-54EF-A4E0-C8785C9233D9}">
  <Class name="AZStd::string" field="value1" value="DynamicFriction" type="{03AAAB3F-5C47-5A66-9EBC-D5FA4DB353C9}"/>
  <Class name="Physics::MaterialPropertyValue" field="value2" version="1" type="{2D9822B6-B4AD-4635-87FD-08F857EEB152}">
    <Class name="float" field="Value" value="0,0000000" type="{EA2C3E90-AFBE-44D4-A90D-FAAF79BAF93D}"/>
  </Class>
</Class>
...
```

**Screenshots/Video**

**Found in Branch**
development

**Commit ID from [o3de/o3de](https://github.com/o3de/o3de) Repository**
6b8fc4d483ad6814ca725fe37127378f57b2afba

**Desktop/Device (please complete the following information):**
 - Device: PC
 - OS: Linux
 - Version Debian 22.04
 - CPU Intel i7-11800H
 - GPU NVIDIA GeForce RTX 3070 Mobile
 - Memory 64MiB

**Additional context**

**Comment by lgleim** [comment]
@o3de/sig-content Discussed in today's SIG Simulation issue triage. This seems like it could be a general issue with the asset compilation pipeline. Adding sig-content for triage.

**Comment by lawsonamzn** [comment]
Yeah, even if you want to support the locale of the UI and stuff, generally, things that are serialized to disk and transported (over git, other things) are in some fixed format, with a fixed locale.  It may be that our serializer/deserializer code has to set locale before reading/writing?

**Comment by monroegm** [comment]
Rerouting to sig/core as this may be a general issue with how values are serialized with different locale configurations

**Comment by nick-l-o3de** [comment]
May be fixed by https://github.com/o3de/o3de/pull/18171

**Comment by michalpelka** [comment]
I've reprduced it with current stabilization branch (without @nick-l-o3de fix).

**Comment by michalpelka** [comment]
Good news @nick-l-o3de, https://github.com/o3de/o3de/pull/18171 fixes that.

**Comment by nick-l-o3de** [comment]
Should we close this, then?

**Comment by michalpelka** [comment]
Let close it.
